### ISSUE TITLE
exporting type instead of interface (configurable)

### ISSUE THREAD

**Issue by dominikdosoudil** [drwpow/openapi-typescript#702]
Hello, I am working on typings that should generate Api interface by typescript definitions generated by openapi-typescript.  For some reason the typing works when `paths`, `components`, ... are exported as a `type` instead of `interface` (`export type paths = {...}`)

I wasn't able to find out what's actually causing the problem however using `type` works fine.

I think that it could be configurable flag when generating the definitions (`paths`, `components`, `operations`).
Something like 
```shell
npx openapi-typescript http://localhost:8000/openapi.json --output apiDefinition.ts --export-type
```

I am up for creating a PR for that if you don't see any problem that could be caused by that. (Maybe there is a reason why it's interface that I don't see)

**Comment by drwpow** [comment]
> I wasn't able to find out what's actually causing the problem however using `type` works fine.

That shouldn’t be the case? `interface` and `type` will perform almost identically, according to TypeScript. I’m against having a flag for this because on top of being unnecessary configuration, it could introduce bugs.

Could you post a reproduction of why an `interface` wasn’t working, along with your schema? I believe there’s an underlying issue here.

**Comment by jonaskello** [comment]
I'm looking for having `type` instead of `interface` too. Although they perform almost identical like @drwpow is saying there are some differences. Specifically I'm bumping into [this typescript issue](https://github.com/microsoft/TypeScript/issues/15300) which documents the difference between interface and type, specifically you can read [this comment](https://github.com/microsoft/TypeScript/issues/15300#issuecomment-332366024) where a typescript team member verifies there are differences.

My specific use-case is that I want to create a mapped type over the generated interface. My mapped type gives a new type for implementing typed handlers for each operation.

Given this generated interface:

```ts
export interface paths {
  "/releases": {
    get: {
      responses: {
        /** A JSON array of releases */
        200: {
          content: {
            "application/json": {
              id?: string;
              name?: string;
            }[];
          };
        };
      };
    };
  };
}
```

I have this mapped type over it:

```ts
export type HandlerFnsFromPathsSpec<T extends PathsSpec, Context> = {
  readonly [K in keyof T]: HandlerFnsFromOperationsSpec<T[K], Context>;
};

export type PathsSpec = {
  readonly [path: string]: OperationsSpec;
};
```

The problem here is that I get this error:

```
Type 'paths' does not satisfy the constraint 'PathsSpec'.   Index signature is missing in type 'paths'.
```

If I just change `export interface paths` to `export type paths =`  it all works. This is consistent with the documented difference between type and interface from above.

So for this reason I'm very interested in having the option to generate `type`  :-).

**Comment by drwpow** [comment]
OK, thanks for explaining more. I think I’m on board with this as a flag.

I’d love someone to submit a PR for this, if possible. It should be a relatively straightforward change, with only having to modify `src/index.ts`. If anyone needs help with the flag addition, happy to assist.

**Comment by mull** [comment]
This can be closed I think? Since 5.2.0
